### PR TITLE
added eslint with typescript and prettier support

### PR DESCRIPTION
**Pull request by jacknugent**
# Feature/Fix Name: 

Added eslint with typescript and prettier support.

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Used [this tutorial](https://www.arden.nl/setting-up-a-gatsby-js-starter-with-type-script-es-lint-prettier-and-pre-commit-hooks) to help install necessary dependencies for eslint, typescript, and prettier to work together.

## Changes
  - Added eslint dependencies to package.json
  - added .eslintrc.js file with configuration

## Type of change

* [ ] Bug fix (non-breaking change which fixes an issue)
* [X] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [X] I have performed a self-review of my own code
* [X] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation
* [X] My changes generate no new warnings
* [ ] Any dependent changes have been merged and published in downstream modules

### New Feature Submissions:

1. [ ] Does your submission pass tests?

2. [ ] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?

* [ ] Have you written new tests for your core changes, as applicable?

* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,12 @@
   "version": "0.1.0",
   "author": "Kyle Mathews <mathews.kyle@gmail.com>",
   "dependencies": {
+    "@typescript-eslint/eslint-plugin": "^1.9.0",
+    "@typescript-eslint/parser": "^1.9.0",
+    "eslint": "^5.16.0",
+    "eslint-config-prettier": "^4.3.0",
+    "eslint-plugin-prettier": "^3.1.0",
+    "eslint-plugin-react": "^7.13.0",
```
Comment (denningk) [inline package.json]: I think these are supposed to be dev dependencies. Looks good otherwise.

Comment (jacknugent) [reply package.json]: Whoops, let me take care of that.

File: package.json [hunk 2]
```diff
@@ -5,6 +5,8 @@
   "version": "0.1.0",
   "author": "Kyle Mathews <mathews.kyle@gmail.com>",
   "dependencies": {
+    "@typescript-eslint/eslint-plugin": "^1.9.0",
+    "@typescript-eslint/parser": "^1.9.0",
```
Comment (denningk) [inline package.json]: I think these two typescript packages are also supposed to be in dev dependencies. Forgot to include these in the last comment.

**Review by denningk — APPROVED** [review]
Looks great. I'll start converting the index to .tsx next.

**Pull request merged**